agent/cache-types/service_dump.go

```diff
@@ -39,6 +39,7 @@ func (c *InternalServiceDump) Fetch(opts cache.FetchOptions, req cache.Request)
 	// allows cached service-discover to automatically read scale across all
 	// servers too.
 	reqReal.AllowStale = true
+	reqReal.ReturnEmptyResultOnUnmodified = true
```

Comment:
I think we would only set this to true if `FetchOptions.LastResult` is non-nil, because we'll need the full response on the first request. But I'm not sure anymore if the cache will benefit from this (related to the other comment).

Reply:
yes, but in any case, the index would be non-valid, so I choose to put it everytime. You might omit it, but it should not change the result anyway